src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -337,6 +339,15 @@ private static boolean hasLineBreakBefore(DetailAST rightCurly) {
      */
     private static final class Details {
 
+        /**
+         * Token types that identify definition tokens.
+         */
+        private static final int[] DEF_TOKENS = {
```

Comment:
I don't think `DEF` accurately descirbes this. We have `METHOD_DEF` for this check but it isn't listed in the array. I forget the exact wording we have used before, but these are basically class definitions. I would rename the list, javadocs, and methods to signify that.

Reply:
I am renaming the list to `TOKENS_WITH_NO_CHILD_SLIST` as all the tokens in this list will never have a child of type `TokenTypes.SLIST` in their corresponding AST.

Reply:
Done